src/main/java/org/medicmobile/webapp/mobile/AppNotificationManager.java

```diff
@@ -0,0 +1,186 @@
+		this.context = context.getApplicationContext();
+		this.appUrl = appUrl;
+		manager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
+		appDataStore = new AppDataStore(this.context);
+		createNotificationChannel();
+	}
+
+	public boolean hasNotificationPermission() {
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+			return ContextCompat.checkSelfPermission(context,
+					Manifest.permission.POST_NOTIFICATIONS)
+					== PackageManager.PERMISSION_GRANTED;
+		}
+		//versions below 13
+		return true;
+	}
+
+	public void requestNotificationPermission(Activity activity) {
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU && !hasNotificationPermission()) {
+			ActivityCompat.requestPermissions(activity,
+					new String[]{Manifest.permission.POST_NOTIFICATIONS}, REQUEST_NOTIFICATION_PERMISSION);
+		}
+	}
+
+	public void startNotificationWorker() {
+		if (hasNotificationPermission()) {
+			PeriodicWorkRequest request = new PeriodicWorkRequest.Builder(
+					NotificationWorker.class,
+					NotificationWorker.WORKER_REPEAT_INTERVAL_MINS,
+					TimeUnit.MINUTES
+			)
+					.addTag(NotificationWorker.NOTIFICATION_WORK_REQUEST_TAG)
+					.build();
+
+			WorkManager.getInstance(context).enqueueUniquePeriodicWork(
+					NotificationWorker.NOTIFICATION_WORK_NAME,
+					ExistingPeriodicWorkPolicy.KEEP,
+					request
+			);
+			log(context, "startNotificationWorker() :: Started Notification Work Manager...");
+		}
+	}
+
+	public void stopNotificationWorker() {
+		appDataStore.saveString(TASK_NOTIFICATIONS_KEY, "[]");
+		WorkManager.getInstance(context).cancelAllWorkByTag(NotificationWorker.NOTIFICATION_WORK_REQUEST_TAG);
+		log(context, "stopNotificationWorker() :: Stopped notification work manager");
+	}
+
+	void cancelAllNotifications() {
+		manager.cancelAll();
+	}
+
+	/**
+	 * @param jsArrayString string notifications sorted by readyAt in descending order
+	 * @throws JSONException throws JSONException
+	 * Method is blocking don't run on UI thread
+	 */
+	public void showNotificationsFromJsArray(String jsArrayString) throws JSONException {
+		JSONArray dataArray = Utils.parseJSArrayData(jsArrayString);
+		if (dataArray.length() == 0) {
+			return;
+		}
+		showMultipleTaskNotifications(dataArray);
+	}
+
+	private void showMultipleTaskNotifications(JSONArray dataArray) throws JSONException {
+		Intent intent = new Intent(context, EmbeddedBrowserActivity.class);
+		intent.setData(Uri.parse(appUrl.concat("/#/tasks")));
+		long startOfDay = getStartOfDay();
+		long latestStoredTimestamp = getLatestStoredTimestamp(startOfDay);
+		for (int i = 0; i < dataArray.length(); i++) {
+			JSONObject notification = dataArray.getJSONObject(i);
+			long readyAt = notification.getLong("readyAt");
+			long endDate = notification.getLong("endDate");
+			if (readyAt > latestStoredTimestamp && endDate >= startOfDay) {
+				int notificationId = (int) (readyAt % Integer.MAX_VALUE);
```

Comment:
Why use the `readyAt` value as the id here instead of just getting the hashcode of the actual id?

```java
int notificationId = notification.getString("_id").hashCode();
```